### PR TITLE
specify testing dependencies

### PR DESCRIPTION
**Pull request by rsyring**
- add new requirments file for all dependencies needed for testing
- update dev guide
- adjust travis to use the requirements specified in the dev guide (eat our own dog food)

### CODE REVIEW

**Comment by rsyring** [comment]
Not sure if you will like the route I went with this.  If not, no problem, just let me know what you would like me to change.

File: docs/devguide.rst [hunk 1]
```diff
@@ -74,10 +74,34 @@ If you want to lint the entire code base run::
...
+Prep your environment for running the tests::
+
+    $ pip install requirements-tests.txt
+
+
+----------------------------------
+Running the Tests - Basic
+----------------------------------
+
+Run all the tests using Nose:
```
Comment (scopatz) [inline docs/devguide.rst]: This line should probably end in a `::`

**Comment by scopatz** [comment]
This looks good to me, mind if I merge?

**Comment by rsyring** [comment]
please do

**Comment by scopatz** [comment]
Thanks a ton!

**Pull request merged**